cmake/module/GenerateMPCode.cmake

```diff
@@ -0,0 +1,32 @@
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+function(target_capnp_sources target)
```

Comment:
This function seems generally useful, so it would be great to move this module to https://github.com/chaincodelabs/libmultiprocess/tree/master/cmake /module, use it upstream to replace hardcoded `add_custom_command` calls, and install it so it could be reused by the bitcoin build.